pkg/generate/app/cmd/newapp.go

```diff
@@ -583,9 +584,16 @@ func (c *AppConfig) detectSource(repositories []*app.SourceRepository) error {
 	return errors.NewAggregate(errs)
 }
 
-func (c *AppConfig) validateEnforcedName() error {
-	if ok, _ := validation.ValidateServiceName(c.Name, false); !ok {
-		return fmt.Errorf("invalid name: %s. Must be an a lower case alphanumeric (a-z, and 0-9) string with a maximum length of 24 characters, where the first character is a letter (a-z), and the '-' character is allowed anywhere except the first or last character.", c.Name)
+func validateEnforcedName(name string) error {
+	if ok, _ := validation.ValidateServiceName(name, false); !ok {
+		return fmt.Errorf("invalid name: %s. Must be an a lower case alphanumeric (a-z, and 0-9) string with a maximum length of 24 characters, where the first character is a letter (a-z), and the '-' character is allowed anywhere except the first or last character.", name)
+	}
+	return nil
+}
+
+func validateOutputImageReference(ref string) error {
+	if _, err := imageapi.ParseDockerImageReference(ref); err != nil {
+		return fmt.Errorf("invalid output image reference: %s", ref)
```

Comment:
Yes, they should. The validation is done against DockerImage because that's a more general pattern. AFAIK, every IST is a parseable DockerImage ref.

What we could do is a more specific validation depending on `--to-docker`, go for that?
(such that `oc new-build <source> --to-docker=false --to repo:80/name/img:tag` would return an error instead of silently ignoring the "repo:80/" part).